### PR TITLE
Remove array properties from types.

### PR DESCRIPTION
**Pull request by skrah**
This removes all array properties from the types.

### CODE REVIEW

File: src/dynd/array.cpp [hunk 1]
```diff
@@ -224,22 +228,77 @@ void nd::array::flag_as_immutable()
...
-    if (!p.is_null()) {
-      return p(eval()); // ToDo: Replace eval() here with an implicit cast
+    const ndt::type &dt = get_dtype();
+
+    switch (dt.get_id()) {
+    case complex_float32_id:
+    case complex_float64_id:
+      return callable_registry[name](eval());
+    case struct_id:
+      return nd::make_field_access_kernel(dt, name)(eval());
```
Comment (skrah) [inline src/dynd/array.cpp]: This is of course inefficient. We could memoize this later in callable_registry.cpp when we update the kernel.

File: src/dynd/types/categorical_type.cpp [hunk 2]
```diff
@@ -469,19 +469,6 @@ struct get_ints_kernel : nd::base_kernel<get_ints_kernel> {
   }
 };
 
-static const std::map<std::string, nd::callable> &categorical_array_properties()
```
Comment (skrah) [inline src/dynd/types/categorical_type.cpp]: Apparently there are no tests for this feature. Should this be handled in array.cpp like for structs?

File: src/dynd/array.cpp (hunk 1 above)
Comment (izaid) [inline src/dynd/array.cpp]: Yep, noted, but fine for now. :)

File: src/dynd/types/categorical_type.cpp (hunk 2 above)
Comment (izaid) [inline src/dynd/types/categorical_type.cpp]: Categorical types are broken, right now. They were too integrated with a variety of things we wanted to remove, so I disabled them. Meaning don't worry for now.

**Comment by izaid** [comment]
For some reason, Travis CI didn't run here. Can you do another push to force it to run?

If the tests pass, I'm happy to merge, but I'll probably do a little follow-up clean-up.

**Comment by skrah** [comment]
Thanks for the review!  I've pushed an unrelated comment change to trigger the build.

**Comment by izaid** [comment]
Merged, thanks!

**Pull request merged**